test/e2e/storage/persistent_volumes-local.go

```diff
@@ -252,6 +253,96 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 		})
 	})
 
+	Context("when pod using local volume with non-existant path", func() {
+		It("should not be able to mount", func() {
+			for _, testVolType := range LocalVolumeTypes {
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol := &localTestVolume{
+					node:            config.node0,
+					hostDir:         "/non-existent/location/nowhere",
+					localVolumeType: testVolType,
+				}
+				By("Creating local PVC and PV")
+				createLocalPVCPV(config, testVol)
+				pod, _ := createLocalPod(config, testVol)
+				var events *v1.EventList
+				selector := fields.Set{
+					"involvedObject.kind":      "Pod",
+					"involvedObject.name":      pod.Name,
+					"involvedObject.namespace": config.ns,
+					"reason":                   "FailedMount",
+				}.AsSelector().String()
+				options := metav1.ListOptions{FieldSelector: selector}
+				events, err := config.client.Core().Events(config.ns).List(options)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(events.Items)).NotTo(Equal(0))
+				Expect(events.Items[0].Message).To(ContainSubstring("MountVolume.SetUp failed"))
+				Expect(events.Items[0].Message).To(ContainSubstring("does not exist"))
+			}
+		})
+	})
+	Context("when pod has nodeName set to a different node than the PV node affinity", func() {
+		It("should not be able to mount", func() {
+			if len(config.nodes.Items) < 2 {
+				framework.Skipf("Runs only when number of nodes >= 2")
+			}
+			for _, testVolType := range LocalVolumeTypes {
+				var testVol *localTestVolume
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol = setupLocalVolumePVCPV(config, testVolType)
+				pod := makeLocalPodWithNodeSelector(config, testVol, config.nodes.Items[1].Name)
+				pod, err := config.client.CoreV1().Pods(config.ns).Create(pod)
+				Expect(err).NotTo(HaveOccurred())
+				framework.WaitForPodRunningInNamespace(config.client, pod)
+				var events *v1.EventList
+				selector := fields.Set{
+					"involvedObject.kind":      "Pod",
+					"involvedObject.name":      pod.Name,
+					"involvedObject.namespace": config.ns,
+					"reason":                   "FailedScheduling",
+				}.AsSelector().String()
+				options := metav1.ListOptions{FieldSelector: selector}
+				events, err = config.client.Core().Events(config.ns).List(options)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(events.Items)).NotTo(Equal(0))
+				Expect(events.Items[0].Message).To(ContainSubstring("MatchNodeSelector"))
+				Expect(events.Items[0].Message).To(ContainSubstring("No nodes are available"))
+				cleanupLocalVolume(config, testVol)
+			}
+		})
+	})
+
+	Context("when pod has NodeSelector different from PV's node affinity", func() {
```

Comment:
There is yet a 3rd way to specify a node in a Pod, using `Pod.NodeName`.

Reply:
done